packages/openapi-generator/src/parser/refs.ts

```diff
@@ -72,6 +104,15 @@ export class OpenApiDocumentRefs {
     );
   }
 
+  /**
+   * If names end with integers it is better to use a separator to easy the mapping between the original and new object.
+   * @param names List of names investigated to find the best separator.
+   * @returns Either '' or '_' depending on the names.
+   */
+  private static getSeperator(names: string[]) {
+    return names.find(name => name.match(/\d+$/)) ? '_' : '';
```

Comment:
Nice to have `_` for schema names ending with numbers. How about using this rule for all cases instead of numbers?

Reply:
I think @marikaner would prefer just a `Class1` instead of a `Class_1` since it is more JS style?

Reply:
@marikaner , here is a question about how should we handle the index pattern when facing name conflicts.
Previous, `Class1` is good enough as JS like style.
However, `Schema400` will become `Schema4001` with the current rule, so we might want to use `Schema400_1` for the schema names ending with number.
Now the question is whether we should align with the same separator `_` or only apply it for number.
I don't have strong opinion, but personally would prefer the alignment a little bit.

Reply:
I use now `_` always but kept it as an optional option.